src/common/dao/replication_job.go

```diff
@@ -136,6 +137,114 @@ func AddRepPolicy(policy models.RepPolicy) (int64, error) {
 }
 
+// AddRepPolicyNg ...
+func AddRepPolicyNg(policy models.RepPolicyNg) (int64, error) {
+	o := GetOrmer()
+	sql := `insert into replication_policy_ng (name, description, creator, src_registry_id, src_namespaces,
+		 dest_registry_id, dest_namespace, override, enabled, cron_str, filters,
+		 replicate_deletion, creation_time, update_time)
+		 values (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?) RETURNING id`
+	params := []interface{}{}
+	now := time.Now()
+
+	params = append(
+		params, policy.Name, policy.Description, policy.Creator, policy.SrcRegistryID, policy.SrcNamespaces,
+		policy.DestRegistryID, policy.DestNamespace, policy.Override, policy.Enabled, policy.Trigger, policy.Filters,
+		policy.ReplicateDeletion, now, now)
+
+	var policyID int64
+	err := o.Raw(sql, params...).QueryRow(&policyID)
+	if err != nil {
+		return 0, err
+	}
+
+	return policyID, nil
+}
+
+// GetPoliciesNg TODO: filter the policies and pagination
+func GetPoliciesNg() ([]*models.RepPolicyNg, error) {
+	o := GetOrmer()
+	sql := `select * from replication_policy_ng where deleted = false`
+
+	var policies []*models.RepPolicyNg
+
+	if _, err := o.Raw(sql).QueryRows(&policies); err != nil {
+		return nil, err
+	}
+
+	return policies, nil
+}
+
+// GetRepPolicyNg ...
+func GetRepPolicyNg(id int64) (*models.RepPolicyNg, error) {
+	o := GetOrmer()
+	sql := `select * from replication_policy_ng where id = ? and deleted = false`
+
+	var policy models.RepPolicyNg
+
+	if err := o.Raw(sql, id).QueryRow(&policy); err != nil {
+		if err == orm.ErrNoRows {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	return &policy, nil
+}
+
+// GetRepPolicyByNameNg ...
+func GetRepPolicyByNameNg(name string) (*models.RepPolicy, error) {
+	o := GetOrmer()
+	sql := `select * from replication_policy_ng where deleted = false and name = ?`
+
+	var policy models.RepPolicy
+
+	if err := o.Raw(sql, name).QueryRow(&policy); err != nil {
+		if err == orm.ErrNoRows {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	return &policy, nil
+}
+
+// UpdateRepPolicyNg TODO: update fields by props
+func UpdateRepPolicyNg(policy *models.RepPolicyNg, props ...string) error {
+	o := GetOrmer()
+
+	// sql := `update replication_policy_ng
+	// 	set name = ?, description = ?, src_registry_id = ?, src_namespaces = ?, dest_registry_id = ?,
+	// 	dest_namespace = ?, override = ?, enabled = ?, cron_str = ?, filters = ?,
+	// 	replicate_deletion = ?, update_time = ? where id = ?`
+
+	// _, err := o.Raw(sql,
+	// 	policy.Name, policy.Description, policy.SrcRegistryID, policy.SrcNamespaces, policy.DestRegistryID,
+	// 	policy.DestNamespace, policy.Override, policy.Enabled, policy.Trigger, policy.Filters,
+	// 	policy.ReplicateDeletion, time.Now(), policy.ID).Exec()
+
+	if policy.ID == 0 {
+		return errors.New("Must set policy ID")
+	}
+
+	// WARNING: If cols is null then update all columns.
+	_, err := o.Update(policy, props...)
+
+	return err
+}
+
+// DeleteRepPolicyNg ...
+func DeleteRepPolicyNg(id int64) error {
```

Comment:
We need to remove the record from database when one policy is deleted, the changes for master branch is submitted https://github.com/goharbor/harbor/pull/6941

Reply:
Hard delete? Why we soft-delete before?